sys/kern/vm_object.c

```diff
@@ -14,38 +14,28 @@ static inline int vm_page_cmp(vm_page_t *a, vm_page_t *b) {
   return a->offset - b->offset;
 }
 
-RB_PROTOTYPE_STATIC(vm_pagetree, vm_page, obj.tree, vm_page_cmp);
-RB_GENERATE(vm_pagetree, vm_page, obj.tree, vm_page_cmp);
-
 vm_object_t *vm_object_alloc(vm_pgr_type_t type) {
   vm_object_t *obj = pool_alloc(P_VMOBJ, M_ZERO);
   TAILQ_INIT(&obj->list);
-  RB_INIT(&obj->tree);
   mtx_init(&obj->mtx, 0);
   obj->pager = &pagers[type];
   obj->ref_counter = 1;
   return obj;
 }
 
-void vm_object_free(vm_object_t *obj) {
```

Comment:
Please do not move the code around if it isn't strictly necessary. That artificially increases amount of changes and confuses a reader! I initially though that you remove `vm_object_free`. Please **always** minimize the changes.

Reply:
In this place moving the code was strictly necessary - to not copy the same code, now in vm_object_free we are calling the vm_object_remove_page_nolock function. Previously, vm_object_free was above this additional function , so the code wouldn't compile, so I had to move the code to some lower position to be below it.